drake/solvers/trajectoryOptimization/PsmTraj.m

```diff
@@ -0,0 +1,445 @@
+classdef PsmTraj < NonlinearProgram
+  % Pseudospectral method
```

Comment:
Thanks for the suggestions, Hongkai. Will rename the class. I also agree with you on the proper hierarchy, I initially did attempt it as a `DirectTrajectoryOptimization` subclass, then ran into a few issues I didn't know a quick fix and rushed into this ugly hack. Will fix it now that there's less time pressure.